lib/assist/assist.go

```diff
@@ -132,6 +133,23 @@ func (a *Assist) GenerateSummary(ctx context.Context, message string) (string, e
 	return a.client.Summary(ctx, message)
 }
 
+// ClassifyMessage takes a user message, a list of categories, and uses the AI
+// mode as a zero shot classifier. It returns an error if the classification
+// result is not a valid class.
+func (a *Assist) ClassifyMessage(ctx context.Context, message string, classes map[string]string) (string, error) {
+	category, err := a.client.ClassifyMessage(ctx, message, classes)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	cleanedCategory := strings.ToLower(strings.Trim(strings.TrimSpace(category), "."))
+	if _, ok := classes[cleanedCategory]; ok {
```

Comment:
I'm afraid this change would break if we create a category name containing another category name. It can also lead to false positives. From what I tested, error rates are quite low. I'd prefer to wait for real data before doing those kinds of optimizations.